### PR TITLE
ZENKO-4824: switch from bitnami-shell to os-shell and debian change

### PR DESCRIPTION
**Pull request by benzekrimaha**
linked Issue: https://scality.atlassian.net/browse/ZENKO-4824

### CODE REVIEW

**Comment by benzekrimaha** [comment]
@bert-e reset

File: VERSION
Comment (francoisferrand) [inline VERSION:1]: we don't want to create a release just yet...

File: solution-base/mongodb/charts/mongodb-sharded/values.yaml
Comment (francoisferrand) [inline solution-base/mongodb/charts/mongodb-sharded/values.yaml:1]: charts **should not** be modified, to minimize maintenance burden (when upgrading the charts):
→ if we override the values anyway, no need to modify the charts
→ if we really need to modify the charts, need to add a `patch` file with the changes, so we can re-apply them after chart upgrade

File: solution-base/build.sh [hunk 1]
```diff
@@ -54,15 +54,15 @@ MONGODB_NAMESPACE=${SOLUTION_ENV}
...
+MONGODB_INIT_IMAGE_NAME="os-shell"
 MONGODB_INIT_IMAGE_TAG=$(yq eval ".mongodb-mongodb-shell.tag" $SOLUTION_BASE_DIR/deps.yaml)
 MONGODB_EXPORTER_IMAGE_NAME="mongodb-exporter"
 MONGODB_EXPORTER_IMAGE_TAG=$(yq eval ".mongodb-exporter.tag" $SOLUTION_BASE_DIR/deps.yaml)
 MONGODB_SHARDED_IMAGE_NAME="mongodb-sharded"
 MONGODB_SHARDED_IMAGE_TAG=$(yq eval ".mongodb-sharded.tag" $SOLUTION_BASE_DIR/deps.yaml)
 MONGODB_SHARDED_EXPORTER_IMAGE_NAME="mongodb-exporter"
 MONGODB_SHARDED_EXPORTER_IMAGE_TAG=$(yq eval ".mongodb-sharded-exporter.tag" $SOLUTION_BASE_DIR/deps.yaml)
-MONGODB_SHARDED_SHELL_IMAGE_NAME="bitnami-shell"
+MONGODB_SHARDED_SHELL_IMAGE_NAME="os-shell"
```
Comment (francoisferrand) [inline solution-base/build.sh]: should be retrieved from `deps.yaml`, like the tag

File: solution-base/build.sh [hunk 2]
```diff
@@ -54,15 +54,15 @@ MONGODB_NAMESPACE=${SOLUTION_ENV}
 MONGODB_REGISTRY=${SOLUTION_REGISTRY}
 MONGODB_IMAGE_NAME="mongodb"
 MONGODB_IMAGE_TAG=$(yq eval ".mongodb.tag" $SOLUTION_BASE_DIR/deps.yaml)
-MONGODB_INIT_IMAGE_NAME="bitnami-shell"
+MONGODB_INIT_IMAGE_NAME="os-shell"
```
Comment (francoisferrand) [inline solution-base/build.sh]: should be retrieved from `deps.yaml`, like the tag

(mapping from "full" image name to ISO is done in `copy_docker_image`, i.e. get only the last part)

**Comment by benzekrimaha** [comment]
@bert-e force_reset

File: solution-base/mongodb/charts/mongodb-sharded/values.yaml
Comment (benzekrimaha) [reply solution-base/mongodb/charts/mongodb-sharded/values.yaml:1]: patch created here : https://github.com/scality/Zenko/pull/2080/commits/a4e595610f4704d8e6a6a044615877e93af857f1#diff-7cd1c3e6b065e85b8733ecc6a676a9fab905dbb7a16360ae0d42bfb998b800f3R1-R15

File: VERSION
Comment (benzekrimaha) [reply VERSION:1]: removed

File: solution-base/build.sh (hunk 1 above)
Comment (benzekrimaha) [reply solution-base/build.sh]: done here : https://github.com/scality/Zenko/pull/2080/commits/a4e595610f4704d8e6a6a044615877e93af857f1#diff-527481e4f2f813ceb9a952c58b56268535ce2bc9fe197644fbbf33f00ca37b79R57

File: solution-base/build.sh (hunk 2 above)
Comment (benzekrimaha) [reply solution-base/build.sh]: done here : https://github.com/scality/Zenko/pull/2080/commits/a4e595610f4704d8e6a6a044615877e93af857f1#diff-527481e4f2f813ceb9a952c58b56268535ce2bc9fe197644fbbf33f00ca37b79R57

File: solution-base/mongodb/patches/values-helper.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
+diff --git a/solution-base/mongodb/charts/mongodb-sharded/values.yaml b/solution-base/mongodb/charts/mongodb-sharded/values.yaml
```
Comment (benzekrimaha) [inline solution-base/mongodb/patches/values-helper.patch:1]: Note for reviewers : please note that this patch will be removed starting the 6.X mongo version as the os shell was introduced here : https://github.com/bitnami/charts/commit/3e976e1fcbe32b73644a7bfd304caf81a81cec4c#diff-bc5147cd6b10d78aec6aac57dfbcb2d5beabcefe05ea3310118a006251efb44a

File: solution-base/mongodb/patches/values-helper.patch [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+--- a/solution-base/mongodb/charts/mongodb-sharded/values.yaml
++++ b/solution-base/mongodb/charts/mongodb-sharded/values.yaml
+@@ -218,8 +218,8 @@ volumePermissions:
+   ##
+   image:
+     registry: docker.io
+-    repository: bitnami/bitnami-shell
+-    tag: 10-debian-10-r293
++    repository: bitnami/os-shell
++    tag: 12-debian-12-r22
```
Comment (ghost) [inline solution-base/mongodb/patches/values-helper.patch:12]: As we are introducing this patch file, I think we should include the other stuff we cusotmize in there as well, like this line

```
  extraArgs: "--collector.diagnosticdata --collector.replicasetstatus --collector.dbstats --collector.topmetrics --compatible-mode"
```

That I believe is not here in the base chart

Comment (benzekrimaha) [reply solution-base/mongodb/patches/values-helper.patch:12]: not sure if we want to introduce this? I am recreating a new patch file as we don't have one for the values.yaml , on integration branches this change will be in the patch file where we have the extraArgs indeed

Comment (benzekrimaha) [reply solution-base/mongodb/patches/values-helper.patch:12]: The `extraArgs` is already in the base chat see here : https://github.com/scality/Zenko/blob/development/2.6/solution-base/mongodb/charts/mongodb-sharded/values.yaml#L1150

**Review by ghost — APPROVED** [review]

**Review by francoisferrand — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Pull request merged**